src/DataProtection/DataProtection/src/KeyManagement/Key.cs

```diff
@@ -92,7 +127,56 @@ public IAuthenticatedEncryptorDescriptor Descriptor
     {
         get
         {
-            return _lazyDescriptor.Value;
+            // We could check for _descriptorException here, but there's no reason to optimize that case
+            // (i.e. by avoiding taking the lock)
+
+            if (_descriptor is not null) // Can only become less null, so losing a race here doesn't matter
```

Comment:
Nit: "less null" is kind of confusing here 😅 I think what this comment is trying to say is that it's not possible to go from a null value to a non-null value for _descriptor since it's only ever reset?

Reply:
I can be more explicit.  You're lucky I didn't just say it was monotonic. 😛